src/bunit.core/ComponentParameterCollectionBuilder.cs

```diff
@@ -314,6 +314,74 @@ public ComponentParameterCollectionBuilder<TComponent> AddUnmatched(string name,
+	/// <param name="valueExpression">Optional value expression, which </param>
+	/// <returns>This <see cref="ComponentParameterCollectionBuilder{TComponent}"/>.</returns>
+	/// <remarks>
+	/// This function is a short-hand form for the following expression:
+	/// <code>p => p
+	/// .Add(c => c.Value, value)
+	/// .Add(c => c.ValueChanged, v => value = v)
+	/// .Add(c => c.ValueExpression, () => value)
+	/// </code>
+	/// With <see cref="Bind{TValue}"/> it looks like this:
+	/// <code>
+	/// p => p.Bind(c => c.Value, value, v => value = v, () => value);
+	/// </code>
+	/// </remarks>
+	public ComponentParameterCollectionBuilder<TComponent> Bind<TValue>(
+		Expression<Func<TComponent, TValue>> parameterSelector,
+		TValue initialValue,
+		Action<TValue> changedAction,
+		Expression<Func<TValue>>? valueExpression = null)
+	{
+		var (parameterName, _, isCascading) = GetParameterInfo(parameterSelector);
+
+		if (isCascading)
+			throw new ArgumentException("Using Bind with a cascading parameter is not allowed.", parameterName);
+		
+		if (changedAction is null)
+        	throw new ArgumentNullException(nameof(changedAction));
+
+		var changedName = $"{parameterName}Changed";
+		var expressionName = $"{parameterName}Expression";
+
+		AssertBindTargetIsCorrect(parameterName, parameterSelector);
+
+		if (!HasPublicParameterProperty(changedName))
+			throw new InvalidOperationException($"The parameter selector '{parameterSelector}' does not resolve to a parameter that has a related parameter with the name {changedName}. This is required for two way binding.");
+
+		AddParameter(parameterName, initialValue);
+		AddParameter(changedName, EventCallback.Factory.Create(changedAction.Target!, changedAction));
+
+		return !HasPublicParameterProperty(expressionName)
+			? this 
+			: AddParameter(expressionName, valueExpression ?? (Expression<Func<TValue>>)(() => initialValue));
+
+		static void AssertBindTargetIsCorrect(string parameterName, Expression<Func<TComponent, TValue>> parameterSelector)
+		{
+			var isBindEventParameter = parameterName.EndsWith("Changed", StringComparison.Ordinal) || parameterName.EndsWith("Expression", StringComparison.Ordinal);
+			var isBindEventType = IsConcreteGenericOf(typeof(TValue), typeof(EventCallback<>)) || IsConcreteGenericOf(typeof(TValue), typeof(Expression<>));
+			if (isBindEventParameter && isBindEventType)
+			{
+				var selectorExpression = parameterSelector.ToString();
+				var possibleSelector = parameterName
+					.Replace("Changed", string.Empty, StringComparison.Ordinal)
+					.Replace("Expression", string.Empty, StringComparison.Ordinal);
+			
+				throw new ArgumentException(
+					$@"The parameter selector {selectorExpression} does not correspond to a valid target for a @bind expression
+If the structure of the component is <MyComponent @bind-Value=""value"" /> call Bind(p => p.Value, ""initial value"", p => p.ValueChanged, v => someVar = v);
```

Comment:
```suggestion
If the structure of the component is <MyComponent @bind-Value=""value"" /> call `Bind(p => p.Value, currentValue, newValue => currentValue= newValue, () => currentValue);`, where `currentValue` is a variable in the test method that has the initial value and gets updated when the `MyComponent.Value` changes.
```

Should it not be something like this?